src/main/java/nwoolcan/model/brewery/production/batch/step/QueryParameter.java

```diff
@@ -0,0 +1,206 @@
+     * @return the {@link Date} that represent the lower bound of the date filter.
+     */
+    public Optional<Date> getStartDate() {
+        return Optional.ofNullable(startDate).map(d -> new Date(d.getTime()));
+    }
+
+    /**
+     * Returns the {@link Date} that represent the upper bound of the date filter.
+     * @return the {@link Date} that represent the upper bound of the date filter.
+     */
+    public Optional<Date> getEndDate() {
+        return Optional.ofNullable(endDate).map(d -> new Date(d.getTime()));
+    }
+
+    /**
+     * Returns true if needs to be sorted by value, false otherwise.
+     * @return true if needs to be sorted by value, false otherwise.
+     */
+    public boolean isSortByValue() {
+        return sortByValue;
+    }
+
+    /**
+     * Returns true if needs to be sorted by date, false otherwise.
+     * @return true if needs to be sorted by date, false otherwise.
+     */
+    public boolean isSortByDate() {
+        return sortByDate;
+    }
+
+    /**
+     * Returns true if sorting in descending order, false otherwise.
+     * @return true if sorting in descending order, false otherwise.
+     */
+    public boolean isSortDescending() {
+        return sortDescending;
+    }
+
+    /**
+     * Sets the parameter type value to filter.
+     * @param parameterType the parameter type value to filter.
+     * @return myself after setting the parameter type value to filter.
+     */
+    public QueryParameter parameterType(final ParameterType parameterType) {
+        this.parameterType = parameterType;
+        return this;
+    }
+
+    /**
+     * Sets the value to use as lower bound for filtering.
+     * @param value the value to use as lower bound for filtering.
+     * @return myself after setting the value to use as lower bound for filtering.
+     */
+    public QueryParameter greaterThanValue(final Number value) {
+        this.greaterThanValue = value;
+        return this;
+    }
+
+    /**
+     * Sets the value to use as upper bound for filtering.
+     * @param value the value to use as upper bound for filtering.
+     * @return myself after setting the value to use as upper bound for filtering.
+     */
+    public QueryParameter lessThanValue(final Number value) {
+        this.lessThanValue = value;
+        return this;
+    }
+
+    /**
+     * Sets the exact value to filter.
+     * @param value the exact value to filter.
+     * @return myself after setting the exact value to filter.
+     */
+    public QueryParameter exactValue(final Number value) {
+        this.exactValue = value;
+        return this;
+    }
+
+    /**
+     * Sets the date to use as lower bound for filtering.
+     * @param startDate the date to use as lower bound for filtering.
+     * @return myself after setting the date to use as lower bound for filtering.
+     */
+    public QueryParameter startDate(final Date startDate) {
```

Comment:
if null an exception has to be thrown. use Objects.requireNonNull() instead

Reply:
setting null is equivalent to not setting anything.
the caller should not be alerted in any way.

Reply:
I don't like it at all. It is responsibility of the controller to check if it is right to call that function. If the controller calls it with null parameter you should throw an error.

Reply:
since java does not provide Properties, this is the closest way to achieve it.
no need to check for null since null does not brake the logic.

Reply:
since java does not provide Properties, this is the closest way to achieve it.
no need to check for null since null does not brake the logic.

Reply:
resolved with #61